x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/analytics_logic.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+import queryString from 'query-string';
+
+import { KibanaLogic } from '../../../shared/kibana';
+import { HttpLogic } from '../../../shared/http';
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { EngineLogic } from '../engine';
+
+import { AnalyticsData, QueryDetails } from './types';
+
+export interface AnalyticsValues extends AnalyticsData, QueryDetails {
+  dataLoading: boolean;
+}
+
+export interface AnalyticsActions {
+  onAnalyticsUnavailable(): void;
+  onAnalyticsDataLoad(data: AnalyticsData): AnalyticsData;
+  onQueryDataLoad(data: QueryDetails): QueryDetails;
+  loadAnalyticsData(): void;
+  loadQueryData(query: string): string;
+}
+
+export const AnalyticsLogic = kea<MakeLogicType<AnalyticsValues, AnalyticsActions>>({
+  path: ['enterprise_search', 'app_search', 'analytics_logic'],
+  actions: () => ({
+    onAnalyticsUnavailable: true,
+    onAnalyticsDataLoad: (data) => data,
+    onQueryDataLoad: (data) => data,
+    loadAnalyticsData: true,
+    loadQueryData: (query) => query,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        loadAnalyticsData: () => true,
+        loadQueryData: () => true,
+        onAnalyticsUnavailable: () => false,
+        onAnalyticsDataLoad: () => false,
+        onQueryDataLoad: () => false,
+      },
+    ],
+    analyticsUnavailable: [
+      false,
+      {
+        onAnalyticsUnavailable: () => true,
+        onAnalyticsDataLoad: (_, { analyticsUnavailable }) => analyticsUnavailable,
```

Comment:
We check `response. analyticsUnavailable` and the listener and call `actions.onAnalyticsUnavailable` if `analyticsUnavailable` is truthy. We only call `onAnalyticsDataLoad` if `analyticsUnavailable` is falsey.

So basically, if `onAnalyticsDataLoad` is triggered, that's saying "analytics are available", since we've already check that condition. So would it simplify things if we just had this be:

```
onAnalyticsDataLoad: () => false
```

Does that makes sense?

Reply:
Awesome catch, thanks Jason!

Any thoughts it might be better to remove the `if (response.analyticsUnavailable)` catch entirely and just let `analyticsUnavailable` get set by the value from the API? 🤔 Which reads more cleanly/simply to you?

Reply:
I think I like the option you proposed better, this makes the reducer feel nice and clean

```
        onAnalyticsUnavailable: () => true,
        onAnalyticsDataLoad: () => true,
        onQueryDataLoad: () => true,
```

Reply:
eaed609

Reply:
Cool! Either way works for me.